Handle scope name version

perebaj opened the pull request:
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->
#### Description

This PR ensures that when Prometheus samples lack the otel_scope_name or otel_scope_version labels, the receiver fills in default values.

It take care of part of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/37277

perebaj committed:
[reciever/prometheusremotewritereciever] handle scope_name and scope_version in receiver

perebaj committed:
adjust valid request test

perebaj committed:
chlog subtext

perebaj committed:
add comment about default scope_name/version

ArthurSens commented on receiver/prometheusremotewritereceiver/receiver.go:
```diff
@@ -225,19 +225,25 @@ func parseJobAndInstance(dest pcommon.Map, job, instance string) {
-	scopeVersion := ls.Get("otel_scope_version")
-	// TODO: If the scope version or scope name is empty, get the information from the collector build tags.
-	// More: https://opentelemetry.io/docs/specs/otel/compatibility/prometheus_and_openmetrics/#:~:text=Metrics%20which%20do%20not%20have%20an%20otel_scope_name%20or%20otel_scope_version%20label%20MUST%20be%20assigned%20an%20instrumentation%20scope%20identifying%20the%20entity%20performing%20the%20translation%20from%20Prometheus%20to%20OpenTelemetry%20(e.g.%20the%20collector%E2%80%99s%20prometheus%2 ...
+	scopeName := prw.settings.BuildInfo.Description
+	scopeVersion := prw.settings.BuildInfo.Version
+
+	if sName := ls.Get("otel_scope_name"); sName != "" {
+		scopeName = sName
+	}
+
+	if sVersion := ls.Get("otel_scope_version"); sVersion != "" {
+		scopeVersion = sVersion
+	}
```
Could we make this a function? We'll repeat this code in all `Add...Datapoints` calls, so we might as well make this reusable instead of duplicated

perebaj replied:
fixed here [91fd77c](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38413/commits/91fd77c11c7bf26732c648fb9e8f93caccaa3af0)

ArthurSens reviewed (commented):
We also have a test case called `duplicated scope name and version` at the top of TestTranslatev2. It would be nice if it were grouped together with the new test at the bottom. While you're at it, could you also add comments with the index positions? It's hard to look at the labelsRef and immediately understand which labels they represent 😛

Also, rereading it, I feel like `duplicated...` sounds confusing 🤔 Where is the duplication? Something like `timeseries with different scopes` sounds more appropriate

perebaj committed:
add extractScopeInfo method

perebaj committed:
reorder test case & add index reference

perebaj commented:
> We also have a test case called `duplicated scope name and version` at the top of TestTranslatev2. It would be nice if it were grouped together with the new test at the bottom. While you're at it, could you also add comments with the index positions? It's hard to look at the labelsRef and immediately understand which labels they represent 😛
> 
> Also, rereading it, I feel like `duplicated...` sounds confusing 🤔 Where is the duplication? Something like `timeseries with different scopes` sounds more appropriate

Fixed [cc705d5](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38413/commits/cc705d5a57e07a2bd9a01f4efab406c2a0683c2e), looking directly in the git diff it's a little bit hard to understand that I just changed the order. Maybe will be easy to understand if you "open the branch"

ArthurSens reviewed (approved):
> Fixed [cc705d5](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38413/commits/cc705d5a57e07a2bd9a01f4efab406c2a0683c2e), looking directly in the git diff it's a little bit hard to understand that I just changed the order. Maybe will be easy to understand if you "open the branch"

Yeah, I did that :) Looks good

The failing test seems to be unrelated to your change.

perebaj committed:
Merge branch 'main' into handle_scope_name_version

perebaj committed:
Merge branch 'main' into handle_scope_name_version
